main.rb

```diff
@@ -0,0 +1,120 @@
+module Enumerable
+  # each
+  def my_each
+    each do |i|
+      yield i
+    end
+  end
+
+  # each_with_index
+  def my_each_with_index
+    result = map do |element, index|
+      yield "#{index}: #{element} "
+    end
+    result
+  end
```

Comment:
- [ ] `my_each_with_index` calls the given block once for each element in self
```
[11,22,31].each.with_index { |val,index| puts "index: #{index} for #{val}" }
index: 0 for 11
index: 1 for 22
index: 2 for 31
=> [11, 22, 31]
```

- [ ] `my_each_with_index` should return an enumerator if no block is given
`p [1, 2, 3].my_each_with_index` returns `#<Enumerator>`